### PR TITLE
Add xcatprobe docs, some docs' status is TODO

### PR DESCRIPTION
**Pull request by xuweibj**
http://10.3.5.4/install/build/html/advanced/probe/index.html

### CODE REVIEW

File: docs/source/advanced/probe/index.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+xCAT probe
+==========
+
+xCAT offers a tool **probe** to help customer to use xCAT.
+
```
Comment (hu-weihua) [inline docs/source/advanced/probe/index.rst]: xCAT offers a new tool suite, which called ``xCAT probe``,  to help customer to probe all the possible issues in xCAT.

File: docs/source/advanced/probe/index.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+You can use ``xcatprobe -l`` to list all valid subcommand, output will be as below  ::
+
```
Comment (hu-weihua) [inline docs/source/advanced/probe/index.rst]: Could you add command line at same time?

    #xcatprobe -l
    Supported sub commands are:
    osdeploy                 Probe for OS provision process, realtime monitor of OS provision process.
     ..........

**Comment by whowutwut** [comment]
If this is not ready by tomorow, we should move to 2.12.4

**Comment by whowutwut** [comment]
@gurevichmark  Can you take a review of the documentation for xcatprobe? 

@xuweibj  Please add a description to this.  Also in the subject, should be `xcatprobe` not `probe` so the team is not confused as to what this pull request is intended for. :-)

**Comment by gurevichmark** [comment]
@hu-weihua The output display for osdeploy probe shown in this document is different from my recommendations that I sent to you on September 22. Perhaps just the code was changed ?

**Comment by gurevichmark** [comment]
A general question about **TODO**. Will it stay like this for now, or are you waiting for other people to add information there under this pull request ?

**Comment by whowutwut** [comment]
Mark good point. Maybe this one is not ready for review yet....

**Comment by hu-weihua** [comment]
@whowutwut  and @gurevichmark , about the **TODO** problem, the reason is ``xCAT Probe`` is a toolkit, the tools are developed by different contributor, so we plan to let the tool owner to complete related document. For example, maybe @gurevichmark is the more suitable person to offer document for tool ``image``.  what do you think?

**Comment by hu-weihua** [comment]
@gurevichmark , thanks for your comments, could you tell me which part of the ``osdeploy`` output is different with your 9.22 comments. We can't login ``C910`` cluster today,  I suppose @xuweibj  pasted the final output of ``osdeploy`` in this document. If you point out where are different, it is helpful. thank you very much.

**Comment by gurevichmark** [comment]
@hu-weihua 

1. In my recommendations for example, I suggested using `osdeploy_probe_report` heading instead of `conclusion_report` and using `All nodes provisioned successfully` instead of `All nodes provision successfully`. 

2. I am also wondering if it is necessary to document the output of `xcatprobe`  here. Perhaps it is better to just give a general description of each subcommand. If we include the output, I think it will be difficult to keep it in sync with the real output. When we change the output of each subcommand to add more function or fix bugs, it will be very easy for the developer to forget that changes are also need to be made for this document page.

3. I do not mind adding some text for `nodecheck`, `osimagecheck` and `image` subcommands. But do you want to do them under this pull request or merge this one and open a new pull request for those changes ? 

4. Do you think a man page needs to be added for the `xcatprobe` command ?

**Comment by hu-weihua** [comment]
@gurevichmark , thanks for your replay.

 1. Sorry for missing your comment " using 'All nodes provisioned successfully' instead of 'All nodes provision successfully'".  I have changed it.  

 2. I discussed with @junxiawang who is the test and first user of ``osdeploy``.  She thought she would be confused if we don't have output example in the document. I know what you concern, we have to sync the output depending on new development. But in my opinion, if it's easy for customer to understand, why we don't do that. On the other hand, if we describe like "the output examples:", is it better?  We told customer the output just is a example. 

 3. Thanks for you would like to contribute the text. This pull request just generate the pages for ``image``, ``nodecheck`` and ``osimagecheck``, after being merged, you can add text in new pull request. We will delete ``TODO`` in the page.

 4. Offer man pages is what we will do next. we plan to offer document first.  @junxiawang needs document to do test.

File: docs/source/advanced/probe/index.rst [hunk 3]
```diff
@@ -1,10 +1,11 @@
 xCAT probe
 ==========
 
-xCAT offers a tool **probe** to help customer to use xCAT.
+xCAT offers a new tool suite, which called **xCAT probe**, to help customer to probe all the possible issues in xCAT.
 
 You can use ``xcatprobe -l`` to list all valid subcommand, output will be as below  ::
```
Comment (hu-weihua) [inline docs/source/advanced/probe/index.rst]: "subcommand" change to "subcommands".

**Comment by gurevichmark** [comment]
I added my comments for grammar and style changes. I will open a pull request soon to add the text for the probes I wrote: image, osimagecheck and nodecheck.

**Comment by xuweibj** [comment]
Hi @gurevichmark , I can not see any comments for grammar and style changes here. Is there something wrong?

**Comment by gurevichmark** [comment]
Sorry, forgot to press the "Submit review" button.

**Comment by xuweibj** [comment]
Hi @gurevichmark , thank you for your comments. I have modified, please help to review again.

**Comment by whowutwut** [comment]
Sorry for the delay response.  I  actually think that since xcatprobe is in beta and really we're looking for feedback, the usage and interface may change and I expect it to change, so I don't think it's very necessary to document it.     I also don't think we should be writing automated testcases against this component since the externals will change often.  

I would like it to get to the point where it becomes integrated into our documentation.  For example, for HW discovery section, perhaps the first step is: 
1) Verify the management node is configured correctly using:  `xcatprobe discovery`

**Comment by daniceexi** [comment]
@whowutwut do you mean to use ``xcatprobe xcatmn`` to check the health of xCAT MN or add the checks what they are done in ``xcatprobe xcatmn`` in the ``xcatprobe discovery``?

**Comment by whowutwut** [comment]
@daniceexi  Yes, sorry, that was a typo.  I fixed it by editing my post.

**Pull request merged**